Several updates in preparation for the 4.8.0 release. (#320)
AxonIQ/reference-guide: axon-framework/deadlines/deadline-managers.md

> Since the non-Pro version of JobRunr has no way to search for deadlines, besides by id, all of the `cancelAll` methods are not implemented for the `JobRunrDeadlineManager`.
> The [JobRunr Pro extension](../../../extensions/jobrunrpro.md) does support those but requires the Pro version of JobRunr.

> **Caveats for the DbScheduler implementation.**
>

> > Db-scheduler has no way to filter out tasks. This means for the implementation of the `cancelAll` methods it will need to serialize all the task data, looping over it. If you have many active deadlines, this might take noticeable time and resources.

Reviewer: ```suggestion
> Db-scheduler has no way to filter out tasks. This means that the `cancelAll` implementation will need to serialize all the task data, looping over it. If you have many active deadlines, this might take noticeable time and resources.
```

Spring Boot users will need to define a `DeadlineManager` bean using one of the available implementations. 

> Spring Boot users who want to use the JobRunr deadline manager can add [`jobrunr-spring-boot-starter`](https://mvnrepository.com/artifact/org.jobrunr/jobrunr-spring-boot-starter) as a dependency. Since this will make a `JobScheduler` bean available, which can be used by the auto configuration to create a `JobRunrDeadlineManager`.
> Alternatively the bean can be configured like this:

Reviewer: ```suggestion
Spring Boot users who want to use the JobRunr deadline manager can add [`jobrunr-spring-boot-starter`](https://mvnrepository.com/artifact/org.jobrunr/jobrunr-spring-boot-starter) as a dependency. Doing so will make a `JobScheduler` bean available, which the auto-configuration can use to create a `JobRunrDeadlineManager`.
Alternatively. the bean can be configured like this:
```

            .spanFactory(spanfactory)
            .build();
}
```

> Spring Boot users who want to use the db-scheduler deadline manager can add [`db-scheduler-spring-boot-starter`](https://mvnrepository.com/artifact/com.github.kagkarlsson/db-scheduler-spring-boot-starter) as a dependency. This will make a `Scheduler` bean available, which can be used by the auto configuration to create a `DbSchedulerDeadlineManager`.
> Alternatively the bean can be configured like this:

Reviewer: ```suggestion
Spring Boot users who want to use the db-scheduler deadline manager can add [`db-scheduler-spring-boot-starter`](https://mvnrepository.com/artifact/com.github.kagkarlsson/db-scheduler-spring-boot-starter) as a dependency. This will make a `Scheduler` bean available, which the auto-configuration can use to create a `DbSchedulerDeadlineManager`.
Alternatively, the bean can be configured like this:
```
